core/card.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Drone IO, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"context"
+	"io"
+
+	"github.com/drone/drone/handler/api/errors"
+)
+
+var (
+	errCardStepInvalid   = errors.New("No Step ID Provided")
+	errCardBuildInvalid  = errors.New("No Build ID Provided")
+	errCardSchemaInvalid = errors.New("No Card Schema Has Been Provided")
+)
+
+type Card struct {
+	Id     int64  `json:"id,omitempty"`
+	Build  int64  `json:"build,omitempty"`
+	Stage  int64  `json:"stage,omitempty"`
+	Step   int64  `json:"step,omitempty"`
+	Schema string `json:"schema,omitempty"`
+}
+
+type CardData struct {
+	Id   int64  `json:"id,omitempty"`
+	Data []byte `json:"card_data"`
+}
+
+// CardStore manages repository cards.
+type CardStore interface {
+	FindCardByBuild(ctx context.Context, build int64) ([]*Card, error)
+	FindCard(ctx context.Context, step int64) (*Card, error)
+	FindCardData(ctx context.Context, id int64) (io.ReadCloser, error)
+	CreateCard(ctx context.Context, card *Card, data io.ReadCloser) error
+	DeleteCard(ctx context.Context, id int64) error
+}
```

Comment:
lets make the naming more consistent with the other stores (no need to include `Card` in the method names)

```
CardStore.Find
CardStore.FindByBuild
CardStore.FindData
CardStore.Create
CardStore.Delete
```

I recommend reading this overview of Go naming conventions, where are a bit different than other languages:
https://medium.com/@lynzt/variable-naming-conventions-in-go-89fe1ef17b0a